### PR TITLE
[FEA] Update to TS v4.5.5

### PR DESCRIPTION
**Pull request by trxcllnt**

### CODE REVIEW

File: modules/cudf/src/series/categorical.ts [hunk 1]
```diff
@@ -166,35 +169,37 @@ export class CategoricalSeries<T extends DataType> extends Series<Categorical<T>
...
+      how: 'left',
+      on: ['categories'],
+      // Pass the memoryResource here because the libcudf semantics are that it should only be
+      // used to allocate memory for the columns that are returned.
+      memoryResource,
+      other: new_df,
+    });
+
+    old_df.drop(['categories']).dispose();
+    new_df.drop(['categories']).dispose();
```
Comment (bryevdv) [inline modules/cudf/src/series/categorical.ts:183]: what's the `drop(...)` needed for?

**Review by bryevdv — APPROVED** [review]

File: modules/cuml/src/utilities/array_utils.ts [hunk 2]
```diff
@@ -35,15 +35,14 @@ export function dataframeToSeries<T extends Numeric, K extends string>(
...
   const nSamples = Math.floor(input.length / nComponents);
+  const indices  = Series.sequence({type: new Int32, init: 0, size: nSamples, step: nComponents});
   let result     = new DataFrame<{[P in number]: T}>({});
   for (let i = 0; i < nComponents; i++) {
-    result = result.assign({
-      [i]:
-        input.gather(Series.sequence({type: new Int32, init: i, size: nSamples, step: nComponents}))
-    });
+    result =
+      result.assign(new DataFrame({[i]: input.gather(Series.new(indices.data.subarray(i)))}));
```
[the remark this answers is not in the document]
Comment (trxcllnt) [reply modules/cuml/src/utilities/array_utils.ts:45]: Oh good point, I missed that

File: modules/cudf/src/series/categorical.ts (hunk 1 above)
Comment (trxcllnt) [reply modules/cudf/src/series/categorical.ts:183]: Dropping the `'categories'` Columns so we don't delete their memory.

Comment (bryevdv) [reply modules/cudf/src/series/categorical.ts:183]: oh I see, a bit counterintuitive to first read

**Review by bryevdv — APPROVED** [review]

**Pull request merged**